### PR TITLE
[Daniel Ogbuti] - Sage - how to build car marketplace dapp using react

### PR DESCRIPTION
**Pull request by dahnny**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 1]
```diff
@@ -0,0 +1,1195 @@
...
+              _carimage,
+              _price,
+              _isUsed,
+              false
+        );
+        carLength++;
+    }
+```
+
+You have to specify the parameters type in the function. In this case, we need to pass the name, description, image, price and isUsed (all with an underscore to differentiate them from the struct values) as parameters to the function. Name the function `setCar` with a visibility of public.
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:167]: THE TYPE OF PARAMETERS(the parameters type)-167 line 1

remove a (with visibility)-line 167- 3

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 2]
```diff
@@ -0,0 +1,1195 @@
...
+The "buyCar" function, which is public and payable, takes in an index of type uint as a parameter.
+
+This function uses the require function to ensure that the cUSD transaction is successful. It then uses the ERC-20 token interface and the stored address to call the transferFrom method to transfer cUSD.
+
+The first parameter is the address of the sender, accessed using the msg.sender method, the second parameter is the recipient of the transaction, which is the owner of the car at the given index, and the final parameter is the price of the car at the given index. 
+
+If the transaction is unsuccessful, an error message will be displayed. If successful, the owner of the car then becomes the sender of the transaction and the status of isBought changes to `true`.
+
+
+In the final section of the smart contract, you would create a function to sell a car and also get the total cars
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:238]: 238 line 2- TO GET THE TOTAL NUMBERS OF CARS

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 3]
```diff
@@ -0,0 +1,1195 @@
...
+## Contract Deployment
+
+To deploy the contract, we would need:
+1. [CeloExtensionWallet]((https://chrome.google.com/webstore/detail/celoextensionwallet/kkilomkmpmkbdnfelcpgckmpcaemjcdh?hl=en))
+2. [ Celo Faucet](https://celo.org/developers/faucet) 
+3. Celo Remix Plugin
+
+Download the Celo Extension Wallet from the Google chrome store using the link above. After doing that, create a wallet, store your key phrase in a very safe place to avoid permanently losing your funds.
+
+After downloading and creating your wallet, you will need to fund it using the Celo Faucet. Copy the address to your wallet, click the link to the faucet above and the paste the address into the text field and confirm.
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:381]: 381 line 2 -correct line->faucet above, paste the address into the text field, and confirm.

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 4]
```diff
@@ -0,0 +1,1195 @@
...
+        .send({ from: address });
+      getCars();
+    } catch (error) {
+      console.log(error);
+    }
+
+  }
+```
+
+You start by creating a new BigNumber instance with the _price argument, and then it's shifting it by ERC20_DECIMALS, this is a way to represent the price in terms of smaller units. Then it's converting the value to a string and storing it in the "price" variable.
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:807]: Not storing but STORES in the line 807 -3

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 5]
```diff
@@ -0,0 +1,1195 @@
...
+
+      getCars();
+    } catch (error) {
+      console.log({ error });
+      alert("Something went wrong");
+    }
+  };
+```
+
+The "buyCar" function takes two arguments: _price, and _index. The function starts by creating a new cUSD contract instance using the ERC20 ABI and cUSD contract address. Then it creates a new BigNumber instance with the _price argument, and then it's shifting it by ERC20_DECIMALS, this is a way to represent the price in terms of smaller units. Then it's converting the value to a string and storing it in the "cost" variable.
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:850]: not storing but STORES line 850 -5

File: blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md [hunk 6]
```diff
@@ -0,0 +1,1195 @@
...
+
+export default AddCar;
+```
+
+Now try to compile your react dapp to see if it is working fine. If it is, you can deploy your dapp on github pages  or netlify.
+You can follow or use this project as a reference to edit yours and get the required files, images e.t.c. <https://github.com/dahnny/cardealer-tutorial>
+
+## Conclusion
+
+Congratulations 🎉, you were able to build your fullstack dapp using solidity and react on the celo smart contract. Great JOB!!
```
Comment (Ishanpathak1) [inline blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md:1187]: Not fullstack but full-stack in line 1187 - 1

**Review by Ishanpathak1 — COMMENTED** [review]
Hey, add the Header image in your .md file, you can download the image from the trello card and then save it in src/datatutorials/showcase/beginner

and then you can mention in your .md file, Thank you, Let me know after you make this changes

**Review by viral-sangani — APPROVED** [review]

**Pull request merged**